python/lsst/ts/electrometer/csc.py

```diff
@@ -255,9 +255,14 @@ async def do_startScan(self, data):
         self.assert_substate(
             substates=[Electrometer.DetailedState.NOTREADINGSTATE], action="startScan"
         )
-        await self.controller.start_scan()
-        await self.report_detailed_state(Electrometer.DetailedState.MANUALREADINGSTATE)
-        self.log.debug("startScan Completed")
+        try:
+            await self.controller.start_scan()
+            await self.report_detailed_state(
+                Electrometer.DetailedState.MANUALREADINGSTATE
+            )
+            self.log.debug("startScan Completed")
+        except Exception:
+            await self.report_detailed_state(Electrometer.DetailedState.NOTREADINGSTATE)
```

Comment:
I gently suggest deleting this method and calling self.evt_detailedState.set_write(detailedState=detailed_state) directly.

I admit that the method does validation, but this is internal and you are already being careful to only write Electrometer.DetailedState values.

Reply:
I'll leave this to Eric.